src/Corax/Pipeline/Token.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Buffers;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using Sparrow;
+using Sparrow.Collections;
+
+namespace Corax.Pipeline
+{
+    // A TokenSequence is a representation of data that is contiguous in memory and therefore can be accessed directly and efficiently. 
+    // Tokenizers may change positions, lengths, types and buffers; but need to return a new TokenSequence in the process.
+    // If the tokenizer, analyzer or filter would not change any of those, they could be reused. 
+    public struct Token
+    {
+        public int Offset;
+        public uint Length;
+        public TokenType Type;
+    }
+
+    public static class TokenTypes
```

Comment:
?? Why duplicate with `TokenType` ?

Reply:
Because we may need to register other types that are compound style. Notice it is a `Flag` type

Reply:
I don't follow, we can have compound flags

Reply:
Those are convenience types, for example: Alphanumeric is Alphabetic | Numeric while some analyzers may need the distinction. I just added a few placeholders there, when we start implementing the analyzing building blocks we will need to add a few there. https://issues.hibernatingrhinos.com/issue/RavenDB-17488

Reply:
So?

```
[Flags]
public enum Foo
{
    None = 0,
    Alpha = 1,
    Numeric  =2 
   Alphanumeric = Alpha | Numeric,
}
```

I don't understand why we have a constant and enum here.

Reply:
That's correct. Forgot that you can do that. Already push a change.